src/profiles_management/helpers/kfam.py

```diff
@@ -0,0 +1,102 @@
+"""Utility module for manipulating KFAM resources."""
+
+import logging
+from typing import List
+
+from lightkube import Client
+from lightkube.generic_resource import GenericNamespacedResource, create_namespaced_resource
+from lightkube.resources.rbac_authorization_v1 import RoleBinding
+
+log = logging.getLogger(__name__)
+
+AuthorizationPolicy = create_namespaced_resource(
+    group="security.istio.io",
+    version="v1beta1",
+    kind="AuthorizationPolicy",
+    plural="authorizationpolicies",
+)
+
+
+def has_kfam_annotations(resource: GenericNamespacedResource) -> bool:
+    """Check if resource has "user" and "role" KFAM annotations."""
+    return (
+        resource.metadata
+        and resource.metadata.annotations
+        and "role" in resource.metadata.annotations
+        and "user" in resource.metadata.annotations
+    )
+
+
+def resource_is_for_profile_owner(resource: GenericNamespacedResource) -> bool:
+    """Check if the resource is for the Profile owner."""
+    if not resource.metadata:
+        return False
+
+    if not has_kfam_annotations(resource):
+        return False
+
+    # AuthorizationPolicy
+    if resource.metadata.name == "ns-owner-access-istio":
+        return True
+
+    # RoleBinding
+    if resource.metadata.name == "namespaceAdmin":
+        return True
+
+    return False
```

Comment:
This way you avoid returning False and giving away incorrect returns, and simplify the method even further.
```suggestion
def resource_is_for_profile_owner(resource: GenericNamespacedResource) -> Optional[bool]:
    if resource.metadata:
      return resource.metadata.name == "ns-owner-access-istio" or resource.metadata.name == "namespaceAdmin"
    return None
```

Reply:
Had to modify slightly as the suggestion wouldn't pass lint checks. Also updated unit tests.

325b7c4

Is there a reason semantically to have `Optional[bool]` and not just `bool`? I guess this is what you mean with "avoid returning False" but why would we not want this?

Reply:
> Is there a reason semantically to have Optional[bool] and not just bool? I guess this is what you mean with "avoid returning False" but why would we not want this?

This is for the case when we return `None` if `resource.metadata` is `None`.

Reply:
Pushed a commit that unified how we handle the functions after our sync. Only `bool` is not returned
1377733

Small heads up that I had to slightly modify the signature, since I unpinned `pyright` and the LSP complained about the types. But should be good now.